Program.cs

```diff
@@ -128,5 +161,78 @@ private static void Pad16(ref List<byte> block, int offset)
                 block.Add((byte)value);
             }
         }
+
+        private static byte[] Domino6Decode(string base64Part)
+        {
+            var decoded = new byte[base64Part.Length];
+            var index = 0;
+
+            for (var i = 0; i < base64Part.Length; i += 4)
+            {
+                int substringLength = i + 4 > base64Part.Length ? base64Part.Length - i : 4;
+                int num = DominoBase64Decode(base64Part.Substring(i, substringLength), substringLength);
+
+                decoded[index++] = Convert.ToByte((num >> 16) & BitMask8Bit);
+                decoded[index++] = Convert.ToByte((num >> 8) & BitMask8Bit);
+                decoded[index++] = Convert.ToByte(num & BitMask8Bit);
+            }
+
+            byte[] salt = decoded.Take(5).ToArray();
+            int byte10 = salt[3] - 4;
+            if (byte10 < 0)
+                byte10 += 256;
+
+            salt[3] = Convert.ToByte(byte10);
+            return salt;
+        }
+
+        private static int DominoBase64Decode(string stringPart, int iterations)
+        {
+            var ret = 0;
+            var i = 1;
+
+            while (i <= iterations)
```

Comment:
I have a hunch that this loop always needs to run 4 times. So far its fine but we will see if its an issue with domino 8 hash. The reason why it doesn't matter is because we are only using the first 5 bytes and discarding all the other decoded bytes
> byte[] salt = decoded.Take(5).ToArray();

Reply:
Yeah, that's just a straight port, so it should be fine as-is. I do vaguely remember reading something about the 8-hash and salts, so it might be an area to keep an eye on.